### ISSUE TITLE
Schannel errors: Recv failure: Connection was reset 

### ISSUE THREAD

**Issue by jeroen** [curl/curl#12112]
We recently upgraded the R bindings on Windows to libcurl 8.3.0. We build on windows using mingw-w64 (gcc-12) with dual ssl (schannel+openssl 3.1.2) but default to the schannel backend.

After the upgrade, one production server that was stable before, is getting frequent errors on requests to various servers:

```
* WARNING: failed to open cookie file ""
  *   Trying 188.114.96.3:443...
  * Connected to jeroen.r-universe.dev (188.114.96.3) port 443
  * schannel: disabled automatic use of client certificate
  * Recv failure: Connection was reset
  * schannel: failed to receive handshake, SSL/TLS connection failed
  * Closing connection
  * schannel: shutting down SSL/TLS connection with jeroen.r-universe.dev port 443
  * Send failure: Connection was reset
  * schannel: failed to send close msg: Failed sending data to the peer (bytes written: -1)
```

Sadly I have difficulty reproducing this error on my own machine; I suspect it only happens under high stress, which makes this even more difficult to debug. 

One thing I did notice is that the failing hosts are often on cloudflare, so they are probably using HTTP2 and TLS 1.3. 

Has anyone perhaps experienced something similar? Googling for the error seems to indicate a recent peak of reports for this error message, but no clear cause. It does smell vaguely like https://github.com/curl/curl/issues/11353

### curl/libcurl version

libcurl 8.3.0

### operating system

Windows 2022

**Comment by jay** [comment]
> Recv failure: Connection was reset

Well, we have to believe this unless you can prove otherwise. If you are able to capture a failed handshake in wireshark that would help.

**Comment by jeroen** [comment]
I have to give up on this for now, because I am unable to get more information from these machines, and can't reproduce it elsewhere. Some hints for if someone runs into this in the future:

 - This problem only appears on our Windows 2022 production servers, I suspect because they are under high stress. I thought maybe a threading issue, so tested using libcurl with different resolvers, disable threading, but no luck.

 - These errors appear frequently for sites that use Cloudflare. I tried downgrading curl but this did not help, old versions of libcurl that worked before had the issue. Cloudflare must have recently enabled something on their networks that triggers a problem. I did try to disable http2 and http3 on my own cloudflare sites, but this did not fix anything.
